python/nammu/view/SyntaxHighlighter.py

```diff
@@ -227,7 +227,19 @@ def syntax_highlight(self, top_line=None, bottom_line=None,
                     color = self.tokencolorlu[tok.type][0]
                     styleline = self.tokencolorlu[tok.type][1]
                 if styleline:
-                    mylength = len(splittext[tok.lineno - 1])
+                    # `styleline` indicates whether we need to apply the style
+                    # to the whole line.
+                    if tok.lineno <= len(splittext):
+                        # Length of the full line.
+                        mylength = len(splittext[tok.lineno - 1])
+                    else:
+                        # For some reason, the line number stored in `tok` is
+                        # larger than the number of lines in `splittext`.  In
+                        # this case, ignore this token and issue a debug
+                        # message.  FIXME: this shouldn't happen.
+                        mylength = 0
```

Comment:
I think `continue` might be better instead of setting this, not so much because you save the call to `setCharacterAttributes`, but because it's not obvious what `mylength = 0` does.

Reply:
`mylength` is the number of characters on which the attributes are set.  With 0, basically nothing should happen.  I'll use `continue`, I think it's cleaner.  I just came up with this alternative after I pushed the branch.